bob/src/link_manager.rs

```diff
@@ -21,28 +24,63 @@ impl LinkManager {
     }
 
     async fn checker_task(factory: Factory, nodes: Arc<[Node]>, period: Duration) {
+        let now = coarsetime::Clock::now_since_epoch().as_secs();
+        Self::checker(
+            &factory,
+            &nodes,
+            Duration::from_millis(FAST_PING_PERIOD_MS),
+            || {
+                let ts = coarsetime::Clock::now_since_epoch().as_secs();
+                ts > now && ts - now > FAST_PING_DURATION_SEC
+            },
+            period.as_millis() as usize / FAST_PING_PERIOD_MS as usize,
+        )
+        .await;
+        Self::checker(&factory, &nodes, period, || false, 1).await;
+    }
+
+    async fn checker(
+        factory: &Factory,
+        nodes: &[Node],
+        period: Duration,
+        should_stop: impl Fn() -> bool,
+        log_iteration_div: usize,
+    ) {
         let mut interval = interval(period);
-        loop {
+        let mut i = 1;
+        while !should_stop() {
+            let log = i % log_iteration_div == 0;
+            if log {
+                i = 1;
+            } else {
+                i += 1;
+            }
             interval.tick().await;
             let mut err_cnt = 0;
             let mut status = String::from("Node status: ");
             for node in nodes.iter() {
                 if let Err(e) = node.check(&factory).await {
-                    error!(
-                        "No connection to {}:[{}] - {}",
-                        node.name(),
-                        node.address(),
-                        e
-                    );
-                    status += &format!("{}{:<10} ", color::Fg(color::Red), node.name());
+                    if log {
+                        error!(
+                            "No connection to {}:[{}] - {}",
+                            node.name(),
+                            node.address(),
+                            e
+                        );
+                        status += &format!("{}{:<10} ", color::Fg(color::Red), node.name());
```

Comment:
Please, remove `color` from text. Most of the time we read logs from text file, not from console